test/weights.jl

```diff
@@ -36,6 +36,23 @@ weight_funcs = (weights, aweights, fweights, pweights)
     @test sum(sa, wv) === 7.0
 end
 
+@testset "$f, setindex!" for f in weight_funcs
+    w = [1., 2., 3.]
+    wv = f(w)
+
+    @test wv[1] == 1.
+    sum(wv) == 6.
+
+    wv[1] = 4.
+    @test wv[1] == 4.
+    @test sum(wv) == 9.
+
+    wv[1:2] = [3., 5.]
+    @test wv[1] == 3.
+    @test wv[2] == 5.
+    @test sum(wv) == 11.
+end
```

Comment:
It would be nice to check 1) that conversion from e.g. `Int` to `Float64` works, 2) that `setindex!` returns the original value (*before* the conversion), and 3) that if assignment fails (e.g. because of conversion error) the sum isn't corrupted.

Otherwise, looks good!